Add usage example to getting started [13416] (#106)
eProsima/DDS-Router: docs/rst/getting_started/usage_example.rst

only forward messages if their associated topics are contained in its ``allowlist``. A ``blocklist`` can also be
specified, on its own or in addition to an ``allowlist``, but we will not be covering this here.

Let us first add only the ``Square`` topic:

.. literalinclude:: ../../resources/getting_started/client-ddsrouter.yaml
    :language: yaml
    :lines: 3-6

We must set ``keyed`` to ``true`` as shape topics use their color attribute as key. See
:ref:`Topic Filtering <user_manual_configuration>` for more details on allowlisting.

Apart from selecting on which topics we wish to send/receive data, we must configure as well the participants that will
ultimately perform communication. Each router instance will contain a :ref:`simple <user_manual_participants_simple>`
and a :ref:`wan <user_manual_participants_wan>` participants. In brief, the simple participants will be in charge of
locally communicating with the corresponding ShapesDemo application, while the wan participants will be the ones
bridging the connection between the two DDS networks.

.. figure:: /rst/figures/shapesdemo_detailed.png

> The only configuration required for simple participants is the DDS ``domain`` identifier.

Reviewer: NIT: Add subsections for LAN and WAN

Author: The example is valid both for LAN and WAN scenarios, with some small differences such as the one pointed out in the initial note.

``listening address``. Note that a unique ``id`` must be set for every wan participant, and this entry should also be
included in ``connection address``. Refer to :ref:`WAN Participant <user_manual_participants_wan>` and
:ref:`WAN Configuration <user_manual_wan_configuration>` for more information. You may also have a look at
:ref:`WAN Example <examples_wan_example>` for a detailed explanation on how to configure this kind of participant.

Following is an example of client and server configuration files:

.. literalinclude:: ../../resources/getting_started/client-ddsrouter.yaml
    :language: yaml

.. literalinclude:: ../../resources/getting_started/server-ddsrouter.yaml
    :language: yaml

Router execution
================
Now, with the configuration files ready, launching a |ddsrouter| instance is as easy as executing the following command:

.. code-block:: bash

    ddsrouter -c config-file.yaml

After setting up both routers, communication between the two ShapesDemo instances should have been established so the
square shape is now visible in the subscriber’s panel.

.. figure:: /rst/figures/shapesdemo_square.png

> |ddsrouter| supports the dynamic addition/deletion of topics at runtime (see
> :ref:`Reload Topics <user_manual_user_interface_reload_topics>`). Let us test this feature by adding the circle topic to
> the allowlist of both routers, and also by removing the square topic (removing this topic from one of the routers’
> allowlist will suffice).

Reviewer: ```suggestion
|ddsrouter| supports the dynamic addition/deletion of topics at runtime (see
:ref:`Reload Topics <user_manual_user_interface_reload_topics>`). Let us test this feature by adding the circle topic to
the allowlist of both routers
Also, by removing the square topic (removing this topic from one of the routers allowlist will suffice) the square data will stop reaching the subscriber..
```

Author: :+1:

.. code-block:: yaml

    allowlist:
    [
        {name: "Circle", type: "ShapeType", keyed: true}
    ]

> After applying these changes, the square should no longer be visible on the subscriber’s side (actually visible but
> frozen), while the circle should.

Reviewer: ```suggestion
After applying these changes, the square should no longer be updated in the subscriber's side (visible but
frozen), while the circle should.
```

Author: :+1:
